src/v/coproc/router.cc

```diff
@@ -322,10 +319,7 @@ errc router::add_source(
               .log = log, .head = topic_offsets(), .scripts = {id}};
             _sources.emplace(ntp, std::move(ts));
         } else {
-            auto& scripts = found->second.scripts;
-            auto id_found = scripts.find(id);
-            vassert(id_found == scripts.end(), "Failed to detect double id");
-            scripts.emplace(id);
+            found->second.scripts.emplace(id);
```

Comment:
In the case an already registered script id wants to be subscripted to data on new ntps, simply add the id to the scripts cache